### PR TITLE
test: migrate to parametrized pytest cases and add scene plausibility checks

### LINKED ISSUE #22: `Scene.result` method yields array of zeros for `num=1`
For `num=1` the `Scene.result` method yields zero for each block.

### LINKED ISSUE #25: Fix all remaining linting errors
Remove ignores in `.ruff.toml` that are not present in upstream.

Currently, this is `D100,D101,D102,D104,N806`

### LINKED ISSUE #28: Add plausibility test for simple fly-by

### LINKED ISSUE #29: Add plausibility test for fly towards and/or away

### LINKED ISSUE #30: Add plausibility test for fly around on a circular trajectory

### PR DESCRIPTION
**Pull request by artpelling**
## Summary

This PR modernizes and expands the test suite by migrating to parametrized pytest case modules and adding plausibility-focused validations for core scene behavior.

## What changed

- Reorganized tests into reusable case-based modules (`cases_*` + `conftest.py`)
- Replaced legacy `test_fly_around.py` with clearer, focused plausibility coverage
- Added new plausibility tests for:
- environment, microphone, source, and trajectory configurations
- `Scene.result` block-size behavior (`num=1` edge case included)
- Minor supporting updates in package/config files to align with the new test setup

## Motivation

The previous tests were harder to extend and did not consistently cover key plausibility and edge-case behavior.
This structure improves maintainability and makes future scenario expansion straightforward.

## Issue linkage

Relates to #17, #27
Closes #22, #25, #28, #29, #30

### CODE REVIEW

File: src/scene_synthesis/directivities.py [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
 import numpy as np
 from traits.api import ABCHasStrictTraits
 
 
 class Directivity(ABCHasStrictTraits):
+    """Base class for directivity patterns."""
+
     def get_coefficients(self, orientation, target_directions):
-        pass
+        """Get directivity coefficients."""
```
[the remark this answers is not in the document]
Comment (jtodev) [reply src/scene_synthesis/directivities.py]: Correct and reasonable improvement; not a current bug but worth making get_coefficients abstract or raising NotImplementedError.

File: src/scene_synthesis/scene.py [hunk 2]
```diff
@@ -59,53 +89,74 @@ def result(self, num=128):
...
-                        receiving_times = np.append(receiving_times, receiving_time)
-                        distances = np.append(distances, distance)
+                        new_receiving_times = np.append(new_receiving_times, receiving_time)
+                        new_distances = np.append(new_distances, distance)
 
                         if source.conv_amp:
-                            radial_Mach = np.dot(source_vel, relative_loc / distance) / c
-                            radial_Machs = np.append(radial_Machs, radial_Mach)
+                            radial_mach = np.dot(source_vel, relative_loc / distance) / c
+                            new_radial_machs = np.append(new_radial_machs, radial_mach)
```
[the remark this answers is not in the document]
Comment (jtodev) [reply src/scene_synthesis/scene.py]: Correct about performance; your current code is logically fine but could be optimized by avoiding np.append in the tight loop.

File: tests/test_plausibility.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    """Test that analytical solution matches synthesis result."""
+
+    # analytical solution
+    def arrival_time_equation(tau_0, tt, src_idx, mic_idx):
+        if scene.sources[src_idx].trajectory is not None:
+            source_pos = scene.sources[src_idx].trajectory.location(tau_0)
+        else:
+            source_pos = scene.sources[src_idx].location[:, np.newaxis]
+        mic_pos = scene.microphones[mic_idx].location[:, np.newaxis]
+        distance_to_mic = np.linalg.norm(source_pos - mic_pos)
```
[the remark this answers is not in the document]
Comment (jtodev) [reply tests/test_plausibility.py:32]: Based on a wrong assumption about Trajectory.location; in this code the method behaves correctly and is validated by tests, so this suggestion is not applicable here.

File: tests/cases_source.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+from tests.cases_trajectory import Trajectories
+
+
+class Sources:
+    """Test cases for acoustic sources."""
+
+    @parametrize_with_cases('trajectories', cases=Trajectories)
+    def case_single(self, trajectories):
+        """Single source test case."""
+        n = 10000
```
[the remark this answers is not in the document]
Comment (jtodev) [reply tests/cases_source.py:16]: Reducing resolution made the plausibility tests fail

**Review by artpelling — COMMENTED** [review]
Great work! This looks really nice. I've added some minor changes. Could you investigate if we can lower the tolerance? Something in the ball park of `1e-11` would be desirable I would say.  

For some trajectories, this leads to failure (10/25 cases fail). Maybe there is something we can fix to improve the error there? Using properly set `rtol` instead of `atol` might help...

**Comment by jtodev** [comment]
I investigated this in `Scene.result()`. The main remaining mismatch came from the linear interpolation of the propagated signal onto the microphone time grid.

I switched that step to cubic spline interpolation (with linear fallback for very short segments), which keeps the block-wise synthesis model unchanged but reduces the analytical mismatch significantly.

With that change, the plausibility tests now pass with `atol=1e-11` for all cases.
